front/app/modules/commercial/insights/admin/containers/Insights/Edit/InputsTable/InputsTable.test.tsx

```diff
@@ -150,6 +177,173 @@ describe('Insights Input Table', () => {
       mockInputData.list[0].relationships.categories.data[0].id
     );
   });
+
+  // Rows selection and actions
+  describe('when no inputs selected', () => {
+    it('has a top-level checkbox to select all inputs in the page', () => {
+      mockLocationData = {
+        pathname: '',
+        query: { category: mockCategoriesData[0].id },
+      };
+      mockInputData = { ...mockInputData, currentPage: 1, lastPage: 1 };
+
+      render(<InputsTable />);
+      // Initially all rows are unchecked
+      expect(
```

Comment:
Love it! You tested this in a really clever way 👍

Reply:
:+1: